src/ty/variant.rs

```diff
@@ -230,6 +230,11 @@ where
         &self.fields
     }
 
+    /// Returns the index of the variant, if specified.
+    pub fn index(&self) -> Option<u8> {
+        self.index
+    }
```

Comment:
What is the difference here from the discriminant?

Reply:
The discriminant is only valid for "C Like" enums. 

If both are specified then the `index` takes precedence for encoding. We still retain the discriminant though for e.g. reconstructing the original type.

Reply:
Ah okay I see. Although we should keep in mind that there is a Rust RFC to allow for discriminants for data variants. Although then `#[scale(index = N)]` would still override those which should be fine.

Reply:
Yes that would still hold true. You are correct that `index` specifically refers to the value from the attribute. And it can be different from the discriminator, but will take precedence for encoding.

Reply:
> keep in mind that there is a Rust RFC

If/when that lands and is stabilised we could deprecate the `#[scale(index = N)]` attribute and only use discriminants. That would be nice (and a breaking change both to us and `parity-scale-codec`).

Reply:
Discriminant removed in https://github.com/paritytech/scale-info/pull/112